misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/distribution/DistributionStepScreen.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react';
+import counterpart from 'counterpart';
+import PropTypes from 'prop-types';
+import { push } from 'connected-react-router';
+
+import { postStepUnPicked } from '../../../api/picking';
+import { updatePickingStepQty } from '../../../actions/PickingActions';
+import ButtonWithIndicator from '../../../components/ButtonWithIndicator';
+import * as CompleteStatus from '../../../constants/CompleteStatus';
+import { toastError } from '../../../utils/toast';
+import { pushHeaderEntry } from '../../../actions/HeaderActions';
+
+class DistributionStepScreen extends Component {
+  handleScanButtonClick = () => {
+    const {
+      stepProps: { dropToLocator, qtyToMove },
+      dispatch,
+      onScanButtonClick,
+    } = this.props;
+    onScanButtonClick();
+
+    dispatch(
+      pushHeaderEntry({
+        location,
+        values: [
+          {
+            caption: counterpart.translate('general.DropToLocator'),
+            value: dropToLocator.barcode,
+          },
+          {
+            caption: counterpart.translate('general.QtyMoved'),
+            value: qtyToMove,
+          },
+        ],
+      })
+    );
+  };
+
+  onUnpickButtonClick = () => {
+    const {
```

Comment:
wonder if we could not move all this to actions and have `onUnpickButton({ wfProcessId ....params })`  and be called from there.